hooks/cascading-scans/hook.test.js

```diff
@@ -859,3 +875,675 @@ test("should copy proper finding ID into annotations", () => {
   }
   )).toBe(true)
 });
+
+test("should merge environment variables into cascaded scan", () => {
+  parentScan.spec.cascades.inheritEnv = true
+  const findings = [
+    {
+      name: "Port 443 is open",
+      category: "Open Port",
+      attributes: {
+        state: "open",
+        hostname: "foobar.com",
+        port: 443,
+        service: "https"
+      }
+    }
+  ];
+
+  parentScan.spec.env = [
+    {
+      "name": "parent_environment_variable_name",
+      "value": "parent_environment_variable_value"
+    }
+  ]
+
+  sslyzeCascadingRules[0].spec.scanSpec.env = [
+    {
+      "name": "rule_environment_variable_name",
+      "value": "rule_environment_variable_value"
+    }
+  ]
+
+  const cascadedScans = getCascadingScans(
+    parentScan,
+    findings,
+    sslyzeCascadingRules
+  );
+
+  const cascadedScan = cascadedScans[0]
+
+  expect(cascadedScans).toMatchInlineSnapshot(`
```

Comment:
I think we should skip these assertions on the `cascadedScans` in these new tests, as these are basically a intermediate result for the purpose of this test, as the actual relevant logic on really happens in `getCascadingScanDefinition`. I was confused for a while looking at this and wondering why there is only on env var in the `cascadedScans` var.

That in itself might be something we should refactor in the future, unifying `getCascadingScans` and `getCascadingScanDefinition` and run all assertions on the actual scan specs. But this doesn't have to happen in this PR.

Reply:
I've updated the getCascadingScans function to return the actual scan definition. Also the tests have been updated, so please take another look.